src/modules/hunting/etcd.py

```diff
@@ -81,66 +50,57 @@ def db_keys_write_access(self):
         data = {
             'value': 'remotely written data'
         }
-        r_secure = "https://{host}:{port}/v2/keys/message".format(host=self.event.host, port=2379)
-        r_not_secure = "http://{host}:{port}/v2/keys/message".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_secure, r_not_secure)
-        if res:
-            self.write_evidence = res.content
-            return True
-        return False
+        r = "{protocol}://{host}:{port}/v2/keys/message".format(host=self.event.host, port=2379, protocol=self.protocol,
+                                                                data=data)
+        self.write_evidence = r.content if r.status_code == 200 and r.content != '' else False
+        return self.write_evidence
 
     def execute(self):
         if self.db_keys_write_access():
             self.publish_event(EtcdRemoteWriteAccessEvent(self.write_evidence))
 
+
 # Passive Hunter
 @handler.subscribe(OpenPortEvent, predicate=lambda p: p.port == 2379)
 class EtcdRemoteAccess(Hunter):
     """Etcd Remote Access
     Checks for remote availability of etcd, version, read access, write access
     """
-    # TODO:
-    # Check the etcd hunter on a remote cluster! (currently everything was checked only at 127.0.0.1:2379)
+
     def __init__(self, event):
         self.event = event
         self.version_evidence = ''
         self.keys_evidence = ''
+        self.protocol = 'https'
 
     def db_keys_disclosure(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to read etcd keys remotely")
-        r_secure = "https://{host}:{port}/v2/keys".format(host=self.event.host, port=2379)
-        r_not_secure = "http://{host}:{port}/v2/keys".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_secure, r_not_secure)
-        if res:
-            self.keys_evidence = res.content
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to read etcd keys remotely")
+        r = requests.get(
```

Comment:
Is there any chance of this (and other calls to request.get/put) failing? Maybe need a try / except to look for connection errors?

Reply:
Yes, there is always a chance for this.
We should catch here requests.exceptions.ConnectionError.